src/libraries/System.Diagnostics.Process/tests/ProcessModuleTests.cs

```diff
@@ -30,7 +30,7 @@ public void TestModuleProperties()
             }
         }
 
-        [Fact]
+        [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
         [SkipOnPlatform(TestPlatforms.iOS | TestPlatforms.tvOS, "libproc is not supported on iOS/tvOS")]
         public void Modules_Get_ContainsHostFileName()
         {
```

Comment:
Wouldn't be better to remove the RemoteExecutor's dependency? e.g:

```cs
[Fact]
[SkipOnPlatform(TestPlatforms.iOS | TestPlatforms.tvOS, "libproc is not supported on iOS/tvOS")]
public void Modules_Get_ContainsHostFileName()
{
    string expectedHostFileName = Path.GetFileName(Process.GetCurrentProcess().MainModule?.FileName);
    ProcessModuleCollection modules = Process.GetCurrentProcess().Modules;
    Assert.Contains(modules.Cast<ProcessModule>(), m => m.FileName.Contains(expectedHostFileName));
}
```

Reply:
I'm not very familiar with this code so I'm not sure if this snippet is a good replacement for all platforms. I agree that this test could be implemented without the RemoteExecutor but its initialization of HostRunnerName seems more nuanced. Just by looking at the history of this test, it seems that it didn't originally use RemoteExecutor but after a few iterations it switched to its implementation of `HostRunnerName`.

Reply:
@Jozkee I went ahead and merged the PR for now. We can continue the discussion and open a new issue or PR if necessary.